src/Serializers/QueryResultSerializer.php

```diff
@@ -56,6 +56,39 @@ public function isSerializerFor( $queryResult ) {
 	}
 
 	/**
+	 * Get display title of a page.
+	 *
+	 * @param int $pageId page ID of page being queried
+	 *
+	 * @return string|bool display title value or false if not found
+	 */
+	public static function getDisplayTitle( $pageId ) {
```

Comment:
should not be public

It being static also causes global state usage (`wfGetDB`). Better use dependency injection. Then again, the functionality added here seems to be out of scope for this class. It is fetching of additional data, not just "serializing" the object that was provided. I can't check right now where this new code would fit unfortunately.